metaflow/plugins/kfp/kfp.py

```diff
@@ -216,75 +228,71 @@ def _command(
         """
         Analogous to batch.py
         """
-        # debug.subcommand is true if this env variable is set:
-        #   export METAFLOW_DEBUG_SUBCOMMAND=1
-        set_debug_command = "set -x" if debug.subcommand else "true"
-        commands = []
-        commands.extend(
-            environment.get_package_commands(code_package_url, pip_install=True)
-            if self.s3_code_package
-            else ["cd " + str(Path(inspect.getabsfile(self.flow.__class__)).parent)]
-        )
-        commands.extend(environment.bootstrap_commands(step_name))
-        commands.append("echo 'Task is starting.'")
-        commands.extend(step_cli)
-        subshell_commands = " && ".join(
-            commands
-        )  # run inside subshell to capture all stdout/stderr
-        # redirect stdout/stderr to separate files, using tee to display to UI
-        redirection_commands = f"> >(tee -a ${RETRY_COUNT}.stdout.log) 2> >(tee -a ${RETRY_COUNT}.stderr.log >&2)"
-
-        # Creating a template to save logs to S3. This is within a function because
-        # datastore_root is not available within the scope of this function, and needs
-        # to be provided in the `step_op` function. f strings (AFAK) don't support
-        # insertion of only a partial number of placeholder strings.
-        def copy_log_cmd(log_file):
-            cp_command = environment.get_boto3_copy_command(
-                s3_path=(
-                    os.path.join(
-                        "$METAFLOW_DATASTORE_SYSROOT_S3",
-                        f"{self.flow.name}/{{run_id}}/{step_name}/${TASK_ID_ENV_NAME}/{log_file}",
-                    )
-                ),
-                local_path=log_file,
-                command="upload_file",
-            )
-            return (
-                f". {STEP_ENVIRONMENT_VARIABLES} "  # for $TASK_ID_ENV_NAME
-                f"&& {cp_command}"
-            )
-
-        # TODO: see datastore get_log_location()
-        #  where the ordinal is attempt/retry count
-        cp_stderr = copy_log_cmd(log_file=f"${RETRY_COUNT}.stderr.log")
-        cp_stdout = copy_log_cmd(log_file=f"${RETRY_COUNT}.stdout.log")
-        cp_logs_cmd = f"{set_debug_command} && {cp_stderr} && {cp_stdout}"
-
         retry_count_python = (
             "import os;"
             'name = os.environ.get("MF_ARGO_NODE_NAME");'
             'index = name.rfind("(");'
             'res = (0 if index == -1 else name[index + 1: -1]) if name.endswith(")") else 0;'
-            f'print("{RETRY_COUNT}=" + str(res))'
+            "print(str(res))"
         )
+        mflog_expr = export_mflog_env_vars(
+            flow_name=self.flow.name,
+            run_id="{run_id}",
+            step_name=step_name,
+            task_id=f"${TASK_ID_ENV_NAME}",
```

Comment:
oh this can be `task_id=TASK_ID_ENV_NAME,`

Reply:
This might be rendering incorrectly... we don't know the task_id until runtime in the foreach case